File: src/main/java/dev/rono/proxychat/listeners/PlayerChatEvent.java
package dev.rono.proxychat.listeners;

import dev.rono.proxychat.ProxyChat;
import dev.rono.proxychat.commands.ChatCommand;
import lombok.var;
import net.md_5.bungee.api.CommandSender;
import net.md_5.bungee.api.connection.ProxiedPlayer;
import net.md_5.bungee.api.event.ChatEvent;
import net.md_5.bungee.api.plugin.Listener;
import net.md_5.bungee.event.EventHandler;

public class PlayerChatEvent implements Listener {
    @EventHandler
    public void onChat(ChatEvent e) {
        if (e.getMessage().startsWith("/")) {
            return;
        }

        for (var command : ProxyChat.getCommands()) {
            var player = (ProxiedPlayer) e.getSender();

            if (command.useCommandPrefix && e.getMessage().startsWith(command.commandPrefix) && player.hasPermission(command.getPermission())) {
                var message = (String) e.getMessage().subSequence(1, e.getMessage().length());
                command.execute(player, message.split(" "));
                e.setCancelled(true);

            } else if (command.toggleUtils.isToggled(player.getUniqueId())) {
                command.execute(player, e.getMessage().split(" "));
                e.setCancelled(true);
            }
        }
    }
}

File: bungee/src/main/kotlin/dev/rono/ProxyChat/Bungee/listeners/PlayerChatEvent.kt
package dev.rono.proxychat.bungee.listeners

import dev.rono.proxychat.bungee.ProxyChatBungee
import net.md_5.bungee.api.connection.ProxiedPlayer
import net.md_5.bungee.api.event.ChatEvent
import net.md_5.bungee.api.plugin.Listener
import net.md_5.bungee.event.EventHandler

class PlayerChatEvent : Listener {
    @EventHandler
    fun onChat(e: ChatEvent) {
        if (e.isCommand || e.isProxyCommand) {
            return
        }

        val player = e.sender as? ProxiedPlayer ?: return

        for (command in ProxyChatBungee.commands) {
            if (command.useCommandPrefix && e.message.startsWith(command.commandPrefix) && player.hasPermission(command.permission)) {
                val message = e.message.substring(command.commandPrefix.length).trim()
                command.handleChat(player, message.split(" ").toTypedArray())
                e.message = "/proxychat"
                return
            } else if (command.toggleUtils.isToggled(player.uniqueId)) {
                command.handleChat(player, e.message.split(" ").toTypedArray())
                e.message = "/proxychat"
                return
            }
        }
    }
}
